In [1]:
import random
# Numpy is a liabrary of numerical mathematics that can make arrays or matrices of numbers
import numpy as np
# import time it function to measure timing of fuctions running
import timeit


In [2]:
# Returns integer between two numbers
random.randint(-90,90)

28

In [3]:
# Returns single floating point decimal between 0 and 1
random.random()

0.7965874926199852

In [4]:
# Algorithym to generate Latitudes between -90 and 90; generate 10 values
x=1
Latitude=[]
while x<11:
    random_Lat=random.randint(-90,90)+random.random()
    Latitude.append(random_Lat)
    x+=1
Latitude


[50.80039161066598,
 37.38352080873219,
 -72.07394767307314,
 2.1842848082874102,
 86.6578888344705,
 52.08414436089677,
 -81.37456992369627,
 -82.09162307638462,
 -27.195017032824236,
 12.186091662617903]

In [5]:
# Randrange function (Upper limit, Upper Limit, step =2)
## Steps by 1 in this case where difference between each number in sequence is 3 or 2
# here below difference between number is in sequence of 3
# default range is 1 if not mentioned
# Gives you integer
random.randrange(-90,90, step=3)


-54

In [6]:
# Uniform function - gives us floating number between two cells 
random.uniform (-90,90)


-66.08058618592311

In [7]:
# Numpy has its own set of random generator function and random liabrary is not needed
# Trailing zero after decimals is not needed
np.random.uniform(-90.000,90.000)

51.46778010303419

In [8]:
np.random.uniform(-90,90)

-23.87406305657511

In [9]:
# We can specify size of the array here in numpy
np.random.uniform(-90.000,90.000,size=50)

array([ 76.11874085,  18.70119865, -58.0030627 ,  36.41284514,
       -84.20064362, -58.13955149, -77.22480045,  86.59305565,
        66.20687198,   5.26969554,   9.76264132, -13.62862889,
       -17.08203604, -75.16651019,  39.35274416,  42.18340334,
        66.8618319 ,  69.98913276,   9.87326039,   5.93574586,
        76.45834441,  24.27509657,  81.57140747,  82.56786225,
       -49.41758113,  16.20588221,  34.91134156,  32.25363825,
        55.57409474,  63.82810163, -11.49073624, -14.99919259,
        13.84287037,  65.05838992,   7.66091583,  79.49653297,
       -11.23155801,  65.53403088, -82.34332848, -35.56705062,
       -45.40108127, -17.09489681,  31.18215992, -44.08573663,
        -6.95003404,  23.63368608,  -0.68004223, -68.13568408,
        12.2378764 ,  31.96180689])

In [12]:
# We can specify size of the array here in numpy
%timeit np.random.uniform(-90.000,90.000,size=1500)

13.4 µs ± 198 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)


The result of %timeit function show 13.4 microseconds - depends upon the RAM of your computer.  

In [15]:
def latitudes(size):
    latitudes = []
    x = 0
    while x < (size):
        random_lat = random.randint(-90, 90) + random.random()
        latitudes.append(random_lat)
        x += 1
    return latitudes
# Call the function with 1500. 
%timeit latitudes(1500)

1.59 ms ± 5.56 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


In [23]:
def longitudes(size):
    longitudes=[]
    x=0
    while x<(size):
        rand_long=random.randint(-180,180)+random.random()
        longitudes.append(rand_long)
        x+=1
    return longitudes
%timeit longitudes(1500)

1.6 ms ± 7.51 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
